webtau-cli/src/main/java/com/twosigma/webtau/cli/Cli.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.cli;
+
+import com.twosigma.webtau.cli.expectation.CliExitCode;
+import com.twosigma.webtau.cli.expectation.CliOutput;
+import com.twosigma.webtau.cli.expectation.CliValidationExitCodeOutputHandler;
+import com.twosigma.webtau.cli.expectation.CliValidationOutputOnlylHandler;
+import com.twosigma.webtau.reporter.TestStep;
+
+import java.util.function.Consumer;
+
+import static com.twosigma.webtau.reporter.IntegrationTestsMessageBuilder.action;
+import static com.twosigma.webtau.reporter.IntegrationTestsMessageBuilder.stringValue;
+import static com.twosigma.webtau.reporter.TokenizedMessage.tokenizedMessage;
+
+public class Cli {
+    public static final Cli cli = new Cli();
+
+    private Cli() {
+    }
+
+    public void run(String command, CliValidationOutputOnlylHandler handler) {
+        cliStep(command, (runResult) -> handler.handle(cliOutput(runResult), cliError(runResult)));
+    }
+
+    public void run(String command, CliValidationExitCodeOutputHandler handler) {
+        cliStep(command, (runResult) -> handler.handle(exitCode(runResult), cliOutput(runResult), cliError(runResult)));
+    }
+
+    private void cliStep(String command, Consumer<ProcessRunResult> validationCode) {
+        TestStep.createAndExecuteStep(null,
+                tokenizedMessage(action("running cli command "), stringValue(command)),
+                () -> tokenizedMessage(action("ran cli clommand"), stringValue(command)),
+                () -> {
+                    ProcessRunResult runResult = ProcessUtils.run(command.split("\\s+"));
```

Comment:
Hmmm, what if someone wants to run `foo.sh "something that should be one arg and is therefore in quotes"`?

Reply:
good point, thanks